File: backend/agents/orchestrator.py
from agents.market import market_research_agent
from agents.branding import branding_agent
from agents.finance import finance_agent
from agents.legalAgent import legal_agent
from agents.business import business_plan_agent
from agents.website import website_agent
from agents.socialMedia import social_media_agent
from agents.competitors import competitor_agent
from agents.pitch import pitchdeck_agent



def run_orchestrator(startup_idea):

    """
    Main AgentX AI Coordinator

    Input:
        startup_idea -> User startup prompt

    Output:
        Complete startup blueprint
    """


    result = {}


    # 1. Market Research Agent

    result["market_research"] = market_research_agent(
        startup_idea
    )



    # 2. Competitor Analysis Agent

    result["competitors"] = competitor_agent(
        startup_idea
    )



    # 3. Branding Agent

    result["branding"] = branding_agent(
        startup_idea
    )



    # 4. Business Plan Agent

    result["business_plan"] = business_plan_agent(
        startup_idea
    )



    # 5. Finance Agent

    result["financial_plan"] = finance_agent(
        startup_idea
    )



    # 6. Legal Agent

    result["legal"] = legal_agent(
        startup_idea
    )



    # 7. Website Builder Agent

    result["website"] = website_agent(
        startup_idea
    )



    # 8. Social Media Agent

    result["marketing"] = social_media_agent(
        startup_idea
    )



    # 9. Pitch Deck Agent

    result["pitchdeck"] = pitchdeck_agent(
        startup_idea
    )



    # Final Response

    final_output = {

        "startup_idea": startup_idea,


        "agents_completed": [

            "Market Research",
            "Competitor Analysis",
            "Branding",
            "Business Plan",
            "Finance",
            "Legal",
            "Website Builder",
            "Social Media",
            "Pitch Deck"

        ],


        "startup_blueprint": result

    }



    return final_output

File: backend/agents/orchestrator.js
const brandingAgent = require("./brandingAgent");
const financeAgent = require("./financeAgent");
const businessAgent = require("./businessAgent");
const websiteAgent = require("./websiteAgent");
const socialAgent = require("./socialAgent");
const competitorAgent = require("./competitor");
const pitchAgent = require("./pitchAgent");
const swotAgent = require("./swotAgent");
const roadmapAgent = require("./roadmap");
const launchAgent = require("./launchAgent");
const legalAgent = require("./legalAgent");
const riskAgent = require("./riskAgent");

/**
 * Executes an agent safely without crashing the workflow.
 */
async function runAgent(name, agentFunction, startupIdea) {
    try {
        const result = await agentFunction(startupIdea);

        return {
            success: true,
            agent: name,
            data: result
        };
    } catch (error) {
        console.error(`❌ ${name} failed:`, error.message);

        return {
            success: false,
            agent: name,
            error: error.message
        };
    }
}

/**
 * Main Orchestrator
 */
async function orchestrator(startupIdea) {

    console.log("=====================================");
    console.log("🚀 Starting AgentX Startup Builder");
    console.log("Idea:", startupIdea);
    console.log("=====================================");

    const startTime = Date.now();

    // ===============================
    // Agent Registry
    // ===============================

    const agents = [
        { name: "Branding Agent", handler: brandingAgent },
        { name: "Finance Agent", handler: financeAgent },
        { name: "Business Plan Agent", handler: businessAgent },
        { name: "Website Agent", handler: websiteAgent },
        { name: "Social Media Agent", handler: socialAgent },
        { name: "Competitor Agent", handler: competitorAgent },
        { name: "Pitch Deck Agent", handler: pitchAgent },
        { name: "SWOT Agent", handler: swotAgent },
        { name: "Roadmap Agent", handler: roadmapAgent },
        { name: "Launch Agent", handler: launchAgent },
        { name: "Legal Agent", handler: legalAgent },
        { name: "Risk Agent", handler: riskAgent }
    ];

    // ===============================
    // Execute All Agents in Parallel
    // ===============================
agents.forEach(agent => {
    console.log(agent.name, typeof agent.handler);
});
    const results = await Promise.all(
        agents.map(agent =>
            runAgent(
                agent.name,
                agent.handler,
                startupIdea
            )
        )
    );

    // ===============================
    // Convert Array -> Object
    // ===============================

    const output = {};

    results.forEach(result => {

        let key = result.agent
            .replace(/\s+/g, "")
            .replace("Agent", "");

        key = key.charAt(0).toLowerCase() + key.slice(1);

        output[key] = result;
    });

    const executionTime = Date.now() - startTime;

    console.log("=====================================");
    console.log("✅ All agents completed");
    console.log(`⚡ Execution Time: ${executionTime} ms`);
    console.log("=====================================");

    return {
        success: true,
        startupIdea,
        generatedAt: new Date().toISOString(),
        executionTime,
        totalAgents: agents.length,
        successfulAgents: results.filter(r => r.success).length,
        failedAgents: results.filter(r => !r.success).length,
        results: output
    };
}

module.exports = {
    generateStartup: orchestrator
};